### PR TITLE
PR builder Jenkins pipeline

### PR DESCRIPTION
**Pull request by mihaiplesa**
Closes https://github.com/brave/brave-browser/issues/2598

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions.

### CODE REVIEW

**Comment by mihaiplesa** [comment]
Almost done, currently it polls for changes in PRs then builds the branch without local merging with target branch. Then runs security, unit and browser tests.

Things left to do are to expose Jenkins webhook endpoint publicly so GithHub can notify Jenkins of changes (instead of polling, if desired) and upgrade the Jenkins and the plugins to fix some bugs in the UI.

**Comment by mihaiplesa** [comment]
this is a test comment

**Comment by mihaiplesa** [comment]
Added stage to create unsigned DMG and store it for future reference.

Added S3-backed sccache support, compilation times have improved.

Switched from `yarn` to `npm`.

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    stages {
+        stage('install') {
+            steps {
+                sh 'npm install'
+            }
+        }
+        stage('init') {
+            when {
+                not {
+                    expression { return fileExists('src/brave/package.json') }
```
Comment (bkero) [inline Jenkinsfile]: If we're okay with this heuristic then I can approve.

**Comment by bbondy** [comment]
pls remember merge label and associated issue's milestone. I did it for this one.

**Pull request merged**